src/main/java/nextstep/optional/ComputerStore.java

```diff
@@ -21,6 +24,11 @@ public static String getVersion(Computer computer) {
     }
 
     public static String getVersionOptional(Computer computer) {
-        return null;
+        Optional<Computer> optionalComputer = Optional.ofNullable(computer);
+
+        return optionalComputer.map(Computer::getSoundcard)
+                .map(Soundcard::getUsb)
+                .map(USB::getVersion)
+                .orElse(UNKNOWN_VERSION);
```

Comment:
지역변수 없이 바로 리턴해도 좋을 것 같아요
```suggestion
        return Optional.ofNullable(computer)
                .map(Computer::getSoundcard)
                .map(Soundcard::getUsb)
                .map(USB::getVersion)
                .orElse(UNKNOWN_VERSION);
```

Reply:
아직은 꺼낸 다음에 수행하는 게 익숙해서 일부러 한 번 꺼내고 변환했었는데요.
역시 후자가 가독성이 더 좋은 것 같네요!